scripts/publish.sh

```diff
@@ -1,15 +1,38 @@
 #!/usr/bin/env bash
 
-set -x
 set -e
 set -o pipefail
 
+export YARN_NPM_AUTH_TOKEN="${YARN_NPM_AUTH_TOKEN:-$NPM_TOKEN}"
+
+YARN_MAJOR="$(yarn --version | sed 's/\..*//' || true)"
+if [[ "$YARN_MAJOR" -ge "3" ]]; then
+  PUBLISH_CMD="yarn npm publish --tag $PUBLISH_NPM_TAG"
+  PACK_CMD="yarn pack --out /tmp/%s-%v.tgz"
+  # install is handled by yarn berry pack/publish
+  INSTALL_CMD=""
+  LOGIN_CMD=""
+else
+  echo "Warning: Did not detect compatible yarn version. This action officially supports Yarn v3 and newer. Falling back to using npm." >&2
+  echo "//registry.npmjs.org/:_authToken=${YARN_NPM_AUTH_TOKEN}" >> $HOME/.npmrc
+  PUBLISH_CMD="npm publish --tag $PUBLISH_NPM_TAG"
+  PACK_CMD="npm pack --pack-destination=/tmp/"
```

Comment:
What is the name of the file that is generated here? Since your original PR we added another step which iterates through `.tgz` files in `/tmp` — each of which is assumed to represent a package — and generate a diff between the latest published version of the package and the newly generated version. So each file in `/tmp` needs to have a unique name to make that work. For the Yarn v3+ command we do this by specifying `--out /tmp/%s-%v.tgz`. Can we do something similar here?

Reply:
Okay perfect. Thanks for checking that.